core/capabilities/syncer_test.go

```diff
@@ -32,9 +38,565 @@ func TestSyncer_CleanStartClose(t *testing.T) {
+
+	var pid ragetypes.PeerID
+	err := pid.UnmarshalText([]byte("12D3KooWBCF1XT5Wi8FzfgNCqRL76Swv8TRU3TiD4QiJm8NMNX7N"))
+	require.NoError(t, err)
+	peer := mocks.NewPeer(t)
+	peer.On("UpdateConnections", mock.Anything).Return(nil)
+	peer.On("ID").Return(pid)
+	wrapper := mocks.NewPeerWrapper(t)
+	wrapper.On("GetPeer").Return(peer)
+
+	nodes := [][32]byte{
+		pid,
+		randomWord(),
+		randomWord(),
+		randomWord(),
+	}
+
+	triggerCapID := randomWord()
+	targetCapID := randomWord()
+	donID := uint32(1)
+	// The below state describes a Workflow DON (AcceptsWorkflows = true),
+	// which isn't public (IsPublic = false), but hosts the
+	// the streams-trigger and write_chain capabilities.
+	// We expect no action to be taken by the syncer.
+	mr := &mockReader{
+		s: state{
+			DONs: []kcr.CapabilityRegistryDONInfo{
+				{
+					Id:               donID,
+					ConfigCount:      uint32(0),
+					F:                uint8(1),
+					IsPublic:         false,
+					AcceptsWorkflows: true,
+					NodeP2PIds:       nodes,
+					CapabilityConfigurations: []kcr.CapabilityRegistryCapabilityConfiguration{
+						{
+							CapabilityId: triggerCapID,
+							Config:       []byte(""),
+						},
+						{
+							CapabilityId: targetCapID,
+							Config:       []byte(""),
+						},
+					},
+				},
+			},
+			IDsToCapabilities: map[capabilityID]kcr.CapabilityRegistryCapability{
+				triggerCapID: {
+					LabelledName:   "streams-trigger",
+					Version:        "1.0.0",
+					CapabilityType: 0,
+				},
+				targetCapID: {
+					LabelledName:   "write-chain_evm_1",
+					Version:        "1.0.0",
+					CapabilityType: 3,
+				},
+			},
+			IDsToNodes: map[p2ptypes.PeerID]kcr.CapabilityRegistryNodeInfo{
+				nodes[0]: {
+					NodeOperatorId:      1,
+					Signer:              randomWord(),
+					P2pId:               nodes[0],
+					HashedCapabilityIds: [][32]byte{triggerCapID, targetCapID},
+				},
+				nodes[1]: {
+					NodeOperatorId:      1,
+					Signer:              randomWord(),
+					P2pId:               nodes[1],
+					HashedCapabilityIds: [][32]byte{triggerCapID, targetCapID},
+				},
+				nodes[2]: {
+					NodeOperatorId:      1,
+					Signer:              randomWord(),
+					P2pId:               nodes[2],
+					HashedCapabilityIds: [][32]byte{triggerCapID, targetCapID},
+				},
+				nodes[3]: {
+					NodeOperatorId:      1,
+					Signer:              randomWord(),
+					P2pId:               nodes[3],
+					HashedCapabilityIds: [][32]byte{triggerCapID, targetCapID},
+				},
+			},
+		},
+	}
+	syncer := newRegistrySyncer(make(services.StopChan), wrapper, registry, dispatcher, lggr, HardcodedDonNetworkSetup{}, mr)
+	require.NoError(t, err)
+
+	// If the DON were public, this would fail with two errors:
+	// - error fetching the capabilities from the registry since they haven't been added
+	// - erroneous calls to dispatcher.SetReceiver, since the call hasn't been registered.
+	err = syncer.sync(ctx)
+	require.NoError(t, err)
```

Comment:
Is there a way to make this a positive assertion? E.g., number of services/receivers/dispatchers running? Negative assertions can sometimes silently stop working.

Reply:
We would probably want to be able to inspect this, too.